src/org/labkey/targetedms/TargetedMSController.java

```diff
@@ -1293,14 +1293,37 @@ public Object execute(QCPlotsDataForm form, BindException errors) throws Excepti
                     filterQCPoints = true;
                 }
             }
-
-            List<QCMetricConfiguration> qcMetricConfigurations = TargetedMSManager.get()
+            List<QCMetricConfiguration> qcMetricConfigurations = TargetedMSManager
                     .getEnabledQCMetricConfigurations(getContainer(), getUser())
                     .stream()
                     .filter(qcMetricConfiguration -> qcMetricConfiguration.getId() == passedMetricId)
                     .collect(Collectors.toList());
+
             List<RawMetricDataSet> rawMetricDataSets = generator.getRawMetricDataSets(getContainer(), getUser(), qcMetricConfigurations, rangeStartDate, form.getEndDate(), form.getSelectedAnnotations(), form.isShowExcluded());
             Map<GuideSetKey, GuideSetStats> stats = generator.getAllProcessedMetricGuideSets(rawMetricDataSets, guideSets.stream().collect(Collectors.toMap(GuideSet::getRowId, Function.identity())));
+
+            // get start date and end date for this qc folder
+            Map<String,Object> qcFolderDateRange = TargetedMSManager.getQCFolderDateRange(getContainer());
+            Date qcStartDate = (Date) qcFolderDateRange.get("startDate");
+            Date qcEndDate = (Date) qcFolderDateRange.get("endDate");
+
+            boolean zoomedRange = qcStartDate.compareTo(rangeStartDate) != 0 || qcEndDate.compareTo(form.getEndDate()) != 0;
```

Comment:
Will we never get to this code if the folder is empty? I'm wondering if the start and end dates could ever be null.

Reply:
@labkey-ankurj did you see this comment about a possible NPE?